ni_measurementlink_service/_drivers/__init__.py

```diff
@@ -16,9 +21,15 @@ def closing_session(session: TSession) -> ContextManager[TSession]:
     Returns:
         A context manager that yields the session and closes it.
     """
-    if isinstance(session, contextlib.AbstractContextManager):
-        # Assume the session yields itself.
-        return session
+    if initialization_behavior == SessionInitializationBehavior.AUTO:
+        # yield the session's context manager.
+        return cast(ContextManager[TSession], session)
+    elif (
+        initialization_behavior
+        == SessionInitializationBehavior.INITIALIZE_SERVER_SESSION_THEN_DETACH
+        or initialization_behavior == SessionInitializationBehavior.ATTACH_TO_SERVER_SESSION
+    ):
+        return contextlib.nullcontext(session)
     elif hasattr(session, "close"):
         return contextlib.closing(session)
```

Comment:
One more thought: for nimi-python and nidaqmx-python, nullcontext and closing are only needed for the two new behaviors. The three existing behaviors are already implemented by the context manager code in nimi-python and nidaqmx-python.

However, using nullcontext and closing for the ATTACH_SERVER_SESSION and INITIALIZE_SERVER_SESSION may eliminate some subtlety for user-provided sessions. Maybe this is a good thing.